sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -415,14 +531,23 @@ public void Dispose()
 
                 _diagnosticSource.Write(_activityName + ".Stop", _diagnosticSourceArgs);
 
+#if NETCOREAPP2_1
                 if (!activity.TryDispose())
+#else
+                if (activity is IDisposable disposable)
+                {
+                    disposable.Dispose();
```

Comment:
Does this also call Stop() on the activity?

Reply:
Yes. it stops the activity and disposes https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.dispose?view=net-7.0#system-diagnostics-activity-dispose

Reply:
This was also a remnant of reflection, I think we can actually just call activity.Dispose() here